README.md

```diff
@@ -325,6 +326,191 @@ const commonProps = {
 
 A warning is displayed when no fallback image is passed. This warning can be disabled in special circumstances. To disable this warning, look in the [warnings section](#warnings).
 
+#### ImgixProvider Component
+
+> New 🪄 ✨
+
+The [`<ImgixProvider>`]() Higher Order Component (HOC), makes its [props](#props) available to any nested `<Imgix>` component in your React application.
+
+For example, by rendering `<ImgixProvider>` at the top level of your application with `imgixParams` defined, all your `<Imgix>` components will have access to the same `imgixParams`.
+
+```jsx
+import React from "react";
+import Imgix, { ImgixProvider } from "react-imgix";
+import HomePage from "./components/HomePage";
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <ImgixProvider imgixParams={{ ar: "16:9" }}>
+          <div className="intro-blurb">{/* ... */}</div>
+          <div className="gallery">
+            <Imgix src="https://assets.imgix.net/examples/pione.jpg" />
+            <Imgix src="https://sdk-test.imgix.net/ساندویچ.jpg" />
+          </div>
+        </ImgixProvider>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+So that the generated HTML looks something like
+
+```html
+<div class="gallery">
+  <img
+    src="https://assets.imgix.net/examples/pione.jpg?auto=format&ar=16%3A9&"
+    ...
+  />
+  <img
+    src="https://sdk-test.imgix.net/%D8%B3%D8%A7%D9%86%D8%AF%D9%88%DB%8C%DA%86.jpg?auto=format&ar=16%3A9&"
+    ...
+  />
+</div>
+```
+
+You can take advantage of this behavior to use 2-step, or partial, URLs with the `<Imgix>` component. By defining the [`domain`](#domain--string-optional) prop on the Provider, it can be made accessible to all nested `<Imgix>` components.
+
+```jsx
+// inside App.jsx
+{
+  /*... */
+}
+<ImgixProvider domain="assets.imgix.net">
+  <div className="intro-blurb">{/* ... */}s</div>
+  <div className="gallery">
+    <Imgix src="/examples/pione.jpg" />
+    <Imgix src="/ساندویچ.jpg" />
+  </div>
+</ImgixProvider>;
+{
+  /*... */
+}
+```
+
+Both the `<Imgix>` components above will access to the `domain` prop from the provider and have their relative `src` paths resolve to the same domain. So that the generated HTML looks something like
```

Comment:
```suggestion
Both the `<Imgix>` components above will access to the `domain` prop from the provider and have their relative `src` paths resolve to the same domain. So that the generated HTML looks something like:
```